New Atmospheric Pipeline

helen-eifert opened the pull request:
Temporary spectrum to remove from radiance scene when DOS-R hasn't been performed.

helen-eifert committed:
Create class

helen-eifert committed:
Update class

helen-eifert committed:
Set up script for classification

Includes file description and package import

helen-eifert committed:
Add files via upload

Brian-Szutu committed:
Addresses issues #46 and #47

Will use a temporary text file if the atmcorr_regr.py output file is missing. Addresses the minor working_dir issue. The check for missing radiance image(s) is moved to the middle to prevent the script from running without the image(s).

Brian-Szutu committed:
Fixes issue #47

Set the working_dir to the folder_dir.

bspitzbart commented:
@helen-eifert @Brian-Szutu Can you check the travis-ci integration reports?  Next to 'continuous-integration/travis-ci/pr' above click Details then Build jobs 16.1 and 16.2.  See what you can clean up style-wise. @iparask, did we change the rule for line length (E501) to 132 chars?  That will eliminate some of the messages to focus on the important ones like:
E: 22, 0: Unable to import 'rasterio' (import-error)

E: 29, 0: Unable to import 'lib.earth_sun_dist' (import-error)

helen-eifert committed:
Starts addressing issue #44

Start to the classification parameter script based on previous structures. Mostly ripped from the refl script, with mostly name adjustments and extraneous lines removed. The new function executed on the image has been adjusted to carry out a sum all bands parameter. This parameter will lead to three classes based on relatively arbitrary thresholds. The output should be a raster of values associated with each of the three classes.

helen-eifert committed:
Addresses Issue #49

Unnecessary code for this script remnant from refl.py

helen-eifert committed:
Subtle math fix

Commit:
Move final raster write outside of sum loop.
Rename to class.py

helen-eifert committed:
Addresses Issue #50

iparask commented on src/cal/atmcorr_specmath.py:
```diff
@@ -67,21 +70,41 @@ def avgs_finder(atmotxt_dir):
-    # Sets a counter variable
+    if not missing_txt:
+        # Opens the .txt document 
+        with open(atmotxt_dir, 'r') as atmo_txt:
```
This is a directory ( `/home/iparask/atmotxt/`) or the path of the actual txt (`/home/iparask/atmotxt/actual_txt.txt`)?

iparask commented on src/cal/atmcorr_specmath.py:
```diff
@@ -67,21 +70,41 @@ def avgs_finder(atmotxt_dir):
+            text = [line.strip().replace("\n", "").split(": ")
+                    for line in atmo_txt]
+        # Closes the file
+        atmo_txt.close()
```
When you do `with` you do not have to close the file at the end. It gets closed automatically

iparask commented on src/cal/atmcorr_specmath.py:
```diff
@@ -178,8 +201,20 @@ def main():
         rad_file = ''
         xml_file = ''
 
+        """
+
+        #UNCOMMENT THIS BLOCK AND REMOVE CHANGE:
+        #folder_dir = working_dir
+        #IN ORDER TO SEARCH THROUGH THE SUBDIRECTORIES OF THE INPUTTED
+        #DIRECTORY
+        
         # Saves the directory of the folder
         folder_dir = os.path.join(working_dir, folder)
+        """
```
Generally in code comments do not go under `"""` I know it is convenient, but it is better if `#` are used. I, for example, add as many `#` as needed.

iparask reviewed (commented):
It looks good to me after a quick look. Well done guys! I'll leave @bspitzbart to decide whether we can move forward with this PR or not.

helen-eifert committed:
Starts addressing Issue  #52

Adds packages necessary for shapefile conversion

helen-eifert committed:
Revisit Issue #42

Library access

helen-eifert committed:
Revisit Issue #42

Library access

helen-eifert committed:
Revisit Issue #42

Old python need

Commit:
Update .flake8rc

helen-eifert committed:
Fixes working directory bugs

Commit:
Fixes issue #52.
Rasterio expects len(src.shape) = 3 and src.shape[0] = len(indexes)
Also need to specify dtype
sum_bands = np.zeros((1,src.height,src.width),dtype=np.float32)

Commit:
fix issue #55

helen-eifert committed:
Addresses Issue #53

Separates polygon part of code and puts it in new script, shp.py. The classification code now outputs multiple 1 band images with the array of 1s and 0s.

helen-eifert committed:
Update naming schemes

Updates the way files are renamed to be more representative and help with further processing.

bspitzbart committed:
Delete tmp

helen-eifert committed:
Fix folder name consistency

a few copied lines still with class in the name instead of shp

Commit:
fix dtype error

Commit:
Merge branch 'experiment/classification' of https://github.com/iceberg-project/LandCover into experiment/classification

Commit:
Revert "Update naming schemes"

This reverts commit fd1d614106a69a0ad247857d01408cdba8cf1162.

helen-eifert committed:
Removes/updates artifacts from old scripts

Commits discussed during call with Brad.

bspitzbart committed:
replace rindex with in (rindex returns exception if 'class' is not found)
add label using regex
